src/driver.toit

```diff
@@ -57,7 +88,23 @@ class Driver:
 
     adapter_ = Adapter_ reader writer logger
 
-    if auto-run: task --background:: run
+    if not disable-auto-run:
+      // Wait for message reciever task to start.
+      // Code moved here and now using a latch to prevent slow startup noticed
+      // in one in 30 odd tests.  (Observed time differences between 25ms
+      // to >800ms for the task startup below.)
+      start := Time.monotonic-us
+      run
+      started := runner-start-latch_.get
+      duration := Duration --us=(Time.monotonic-us - start)
```

Comment:
You are only interested in the `ms`. No need to create a duration.
But you can also (assuming:
```
duration := Duration.of:
  latch := run
  latch.get
```
or, for a semaphore
```
duration := Duration.of:
  semaphore := run
  semaphore.down
```
Note that you don't need a latch